src/udev/udevadm-info.c

```diff
@@ -662,41 +798,96 @@ static int print_tree(sd_device* below) {
         return 0;
 }
 
+static int parse_key_value_argument(const char *s, char **key, char **value) {
+        _cleanup_free_ char *k = NULL, *v = NULL;
+        int r;
+
+        assert(s);
+        assert(key);
+        assert(value);
+
+        r = extract_many_words(&s, "=", EXTRACT_DONT_COALESCE_SEPARATORS, &k, &v, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse key/value pair %s: %m", s);
+        if (r < 2)
+                return log_error_errno(SYNTHETIC_ERRNO(r), "Missing '=' in key/value pair %s.", s);
```

Comment:
Maybe, better to check that the input string does not contain multiple `=`, e.g. `KEY=VAL=FOO` ?

Reply:
Would that be a problem? AFAIK there's nothing preventing the kernel from putting '=' in property values.

Reply:
`r` is not a valid error, first arg should be changed to `SYNTHETIC_ERRNO(EINVAL)` or so